test/e2e/clusterclass_rollout.go

```diff
@@ -1102,6 +1098,8 @@ func getClusterObjects(ctx context.Context, g Gomega, clusterProxy framework.Clu
 		InfrastructureMachineTemplateByMachineDeployment: map[string]*unstructured.Unstructured{},
 		BootstrapConfigByMachine:                         map[string]*unstructured.Unstructured{},
 		InfrastructureMachineByMachine:                   map[string]*unstructured.Unstructured{},
+		BootstrapConfigByMachinePool:                     map[string]*unstructured.Unstructured{},
+		InfrastructureMachinePoolByMachinePool:           map[string]*unstructured.Unstructured{},
```

Comment:
This just sets them to an empty map. What I meant is that we have to iterate through the MachinePools, get the corresponding BootstrapConfig and InfrastructureMachinePool objects and add them to these maps.

Similar to what we do for MDs in l. 1139-1145